src/later.cpp

```diff
@@ -69,19 +73,67 @@ bool at_top_level() {
   return nframe == 0;
 }
 
-// The queue of user-provided callbacks that are scheduled to be
-// executed.
-CallbackRegistry callbackRegistry;
+
+// Each callback registry represents one event loop. Note that traversing and
+// modifying callbackRegistries should always happens on the same thread, to
+// avoid races.
+std::map<int, boost::shared_ptr<CallbackRegistry> > callbackRegistries;
+
+// [[Rcpp::export]]
+bool existsCallbackRegistry(int loop) {
+  ASSERT_MAIN_THREAD()
+  return (callbackRegistries.find(loop) != callbackRegistries.end());
+}
+
+// [[Rcpp::export]]
+bool createCallbackRegistry(int loop) {
+  ASSERT_MAIN_THREAD()
+  if (existsCallbackRegistry(loop)) {
+    Rcpp::stop("Can't create event loop %d because it already exists.", loop);
+  }
+  callbackRegistries[loop] = boost::make_shared<CallbackRegistry>();
+  return true;
+}
+
+// Gets a callback registry by ID. If registry doesn't exist, it will be created.
+boost::shared_ptr<CallbackRegistry> getCallbackRegistry(int loop) {
+  ASSERT_MAIN_THREAD()
+  // TODO: clean up
```

Comment:
Do you know what this is referring to?

Reply:
Nope. You wrote it. ;)
https://github.com/r-lib/later/blame/e43e03ee5bcbc9f412c17c63a0bc4f32c7b7a523/src/later.cpp#L101